lib/Tomcat/Utils.pm

```diff
@@ -68,20 +71,28 @@ EOF
 }
 
 
-# Access the tomcat web application manager
+# Check Servlet, JSP and Websocket, the example files can be accessed, login with admin via tomcat manager
 sub tomcat_manager_test() {
-    my ($self) = shift;
-
-    $self->firefox_open_url('localhost:8080/manager');
-    assert_screen('tomcat-manager-authentication', 180);
-    type_string('admin');
-    send_key('tab');
-    type_string('admin');
-    assert_and_click('tomcat-OK-autentication');
-    wait_still_screen(2);
-    assert_and_click('tomcat-click-save-login');
-    assert_screen('tomcat-web-application-manager', TIMEOUT);
+    select_serial_terminal;
+
+    # sometimes we have network issue (cannot connect to localhost)
+    systemctl('restart network.service');
+    assert_script_run('ping -c 4 localhost');
+
+    record_info('curl examples of Servelt, JSP and Websocket');
+    # curl Servlet examples and login with authentification
+    assert_script_run('curl --connect-timeout 20 --user admin:admin --output servelets 127.0.0.1:8080/examples/servlets', 90);
```

Comment:
after pulling the network from under the tomcat server (above, network restart) I'm actually surprised the tomcat works reliably here (except if the service managed to have restarted thanks to systemd chaining too)

Reply:
yes, my goal is avoid sporadic issue which bothers us in future ;)